crates/proof-of-sql/src/base/commitment/naive_commitment.rs

```diff
@@ -0,0 +1,157 @@
+
+    fn add(self, rhs: Self) -> Self::Output {
+        let mut new_self = self.clone();
+        new_self.add_assign(rhs);
+        new_self
+    }
+}
+
+impl Sub for NaiveCommitment {
+    type Output = NaiveCommitment;
+
+    fn sub(self, rhs: Self) -> Self::Output {
+        let mut new_self = self.clone();
+        new_self.sub_assign(rhs);
+        new_self
+    }
+}
+impl Neg for NaiveCommitment {
+    type Output = NaiveCommitment;
+
+    fn neg(self) -> Self::Output {
+        Self(self.0.iter().map(|s| s.neg()).collect())
+    }
+}
+
+impl SubAssign for NaiveCommitment {
+    fn sub_assign(&mut self, rhs: Self) {
+        self.add_assign(rhs.neg())
+    }
+}
+
+impl AddAssign for NaiveCommitment {
+    fn add_assign(&mut self, rhs: Self) {
+        if self.0.len() < rhs.0.len() {
+            self.0
+                .extend((self.0.len()..rhs.0.len()).map(|_i| TestScalar::ZERO));
+        }
+        self.0
+            .iter_mut()
+            .zip(rhs.0)
+            .for_each(|(lhs, rhs)| *lhs += rhs);
+    }
+}
+
+impl PartialEq for NaiveCommitment {
+    fn eq(&self, other: &Self) -> bool {
+        match self.0.len().cmp(&other.0.len()) {
+            std::cmp::Ordering::Less => {
+                let mut extended_self = self.0.clone();
+                extended_self.extend((self.0.len()..other.0.len()).map(|_i| TestScalar::ZERO));
+                extended_self == other.0
+            }
+            std::cmp::Ordering::Equal => self.0 == other.0,
+            std::cmp::Ordering::Greater => {
+                let mut extended_other = other.0.clone();
+                extended_other.extend((other.0.len()..self.0.len()).map(|_i| TestScalar::ZERO));
+                extended_other == self.0
+            }
+        }
+    }
+}
+
+impl core::ops::Mul<NaiveCommitment> for TestScalar {
+    type Output = NaiveCommitment;
+    fn mul(self, rhs: NaiveCommitment) -> Self::Output {
+        NaiveCommitment(rhs.0.iter().map(|s| self * *s).collect())
+    }
+}
+impl core::ops::Mul<TestScalar> for NaiveCommitment {
+    type Output = NaiveCommitment;
+    fn mul(self, rhs: TestScalar) -> Self::Output {
+        NaiveCommitment(self.0.iter().map(|s| rhs * *s).collect())
+    }
+}
+impl core::ops::Mul<&NaiveCommitment> for TestScalar {
+    type Output = NaiveCommitment;
+    fn mul(self, rhs: &NaiveCommitment) -> Self::Output {
+        NaiveCommitment(rhs.0.iter().map(|s| self * *s).collect())
+    }
+}
+impl core::ops::Mul<TestScalar> for &NaiveCommitment {
+    type Output = NaiveCommitment;
+    fn mul(self, rhs: TestScalar) -> Self::Output {
+        NaiveCommitment(self.0.iter().map(|s| rhs * *s).collect())
+    }
+}
+
+impl Commitment for NaiveCommitment {
+    type Scalar = TestScalar;
+    type PublicSetup<'a> = ();
+
+    fn compute_commitments(
+        commitments: &mut [Self],
+        committable_columns: &[CommittableColumn],
+        offset: usize,
+        _setup: &Self::PublicSetup<'_>,
+    ) {
```

Comment:
There are unit tests that currently exist that I believe are meant to be generic that test compute_commitments. Because they will be using TestScalar in the next PR, should I just pull those out of wherever they are (column_commitments) and put them here?